wgpu-core/src/command/render.rs

```diff
@@ -2408,7 +2408,9 @@ fn set_push_constant(
 fn set_scissor(state: &mut State, rect: Rect<u32>) -> Result<(), RenderPassErrorInner> {
     api_log!("RenderPass::set_scissor_rect {rect:?}");
 
-    if rect.x + rect.w > state.info.extent.width || rect.y + rect.h > state.info.extent.height {
+    if rect.x.saturating_add(rect.w) > state.info.extent.width
+        || rect.y.saturating_add(rect.h) > state.info.extent.height
+    {
```

Comment:
thought: I don't think it's important, but the approach I've normally taken is to use checked arithmetic (i.e., `{integer}::checked_add(…)`. I don't think there's a significant difference here, though, mostly just something I wanted to share.

Reply:
Thanks! Yeah `checked_add` would work fine too but it would be slightly more verbose to split out the two `None` cases here. It would allow us to handle an extent dimension of exactly `u32::MAX` (i.e., saturating is technically wrong in that case), but I don't think we'll need to worry about that.